implementations/elixir/ockam/ockam_healthcheck/lib/healthcheck/healthcheck.ex

```diff
@@ -67,6 +73,72 @@ defmodule Ockam.Healthcheck do
     end)
   end
 
+  def check_api_endpoint(target, timeout) do
+    %{
+      host: host,
+      path: path,
+      method: method,
+      body: body,
+      port: port,
+      api_worker: api_worker,
+      healthcheck_worker: healthcheck_worker
+    } = target
+
+    body = decode_body(body)
+
+    with_tcp(host, port, fn conn ->
+      with_channel(conn, api_worker, fn channel ->
+        {:ok, me} = Ockam.Node.register_random_address()
+        request_id = Enum.random(1..65_534)
```

Comment:
Depending on how this will be used downstream, we could also consider `System.monotonic_time` or `System.unique_integer`.